*** cow_wrap.c	2012-12-06 22:31:05.000000000 -0500
--- cow_wrap_patched.c	2012-12-06 22:30:53.000000000 -0500
***************
*** 9,14 ****
--- 9,16 ----
   * ----------------------------------------------------------------------------- */
  
  #define SWIGLUA
+ #define luaL_reg luaL_Reg
+ #include "../include/cow.h"
  
  /* -----------------------------------------------------------------------------
   *  This section contains generic SWIG labels for method/variable
***************
*** 5002,5017 ****
  SWIGEXPORT int SWIG_init(lua_State* L)
  {
    int i;
!   /* start with global table */
!   lua_pushvalue(L,LUA_GLOBALSINDEX);
    /* SWIG's internal initalisation */
    SWIG_InitializeModule((void*)L);
    SWIG_PropagateClientData();
!   /* add a global fn */
!   SWIG_Lua_add_function(L,"swig_type",SWIG_Lua_type);
!   SWIG_Lua_add_function(L,"swig_equals",SWIG_Lua_equal);
    /* begin the module (its a table with the same name as the module) */
    SWIG_Lua_module_begin(L,SWIG_name);
    /* add commands/functions */
    for (i = 0; swig_commands[i].name; i++){
      SWIG_Lua_module_add_function(L,swig_commands[i].name,swig_commands[i].func);
--- 5004,5023 ----
  SWIGEXPORT int SWIG_init(lua_State* L)
  {
    int i;
!   lua_getglobal(L, "package");
!   lua_getfield(L, -1, "loaded");
! 
    /* SWIG's internal initalisation */
    SWIG_InitializeModule((void*)L);
    SWIG_PropagateClientData();
! 
    /* begin the module (its a table with the same name as the module) */
    SWIG_Lua_module_begin(L,SWIG_name);
+ 
+   /* add a function to module */
+   SWIG_Lua_add_function(L,"swig_type",SWIG_Lua_type);
+   SWIG_Lua_add_function(L,"swig_equals",SWIG_Lua_equal);
+ 
    /* add commands/functions */
    for (i = 0; swig_commands[i].name; i++){
      SWIG_Lua_module_add_function(L,swig_commands[i].name,swig_commands[i].func);
***************
*** 5037,5044 ****
    /* invoke user-specific initialization */
    SWIG_init_user(L);
    /* end module */
!   lua_pop(L,1);  /* tidy stack (remove module table)*/
!   lua_pop(L,1);  /* tidy stack (remove global table)*/
    return 1;
  }
  
--- 5043,5053 ----
    /* invoke user-specific initialization */
    SWIG_init_user(L);
    /* end module */
! 
!   lua_pop(L,1);  /* tidy stack (remove module table) */
!   lua_pop(L,1);  /* tidy stack (remove loaded table) */
!   lua_pop(L,1);  /* tidy stack (remove package table) */
! 
    return 1;
  }
  
